### PR TITLE
Update Secvisogram to match current schema version

### LINKED ISSUE #6: Add product_identification_helper
https://github.com/oasis-tcs/csaf/pull/197 implements the `product_indentification_helper`. This should be supported by Secvisogram.

### PR DESCRIPTION
**Pull request by tschmidtb51**
This is a broader update to match the schema version of [oasis-tcs/csaf](https://github.com/oasis-tcs/csaf) of branch [editor-revision-2021-03-23](https://github.com/oasis-tcs/csaf/blob/editor-revision-2021-03-23/csaf_2.0/json_schema/csaf_json_schema.json). It includes:
- full support of "Product Identification Helper" (oasis-tcs/csaf#197, fix #6)
- parts of "Alphabetical order of properties" (oasis-tcs/csaf#194, #7)
- minor refactoring

### CODE REVIEW

**Review by pdamian — COMMENTED** [review]
Primarily tested the "Product Identification Helper" functionality. Could not identify any issues.

Could be helpful though for the end user to see examples of valid fields, especially for CPEs (instead of just the complex regex that needs to be satisfied). Examples would probably needed to be added to the CSAF JSON schema.

**Comment by tschmidtb51** [comment]
@pdamian: I finished updating the docs. Could you please have a look again?

**Review by pdamian — COMMENTED** [review]
The docs look good to me as well.

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
   },
   "repository": {
     "type": "git",
-    "url": "git@gitlab-ext.exxcellent.de:bsi/secvisogram.git"
+    "url": "git@github.com:secvisogram/secvisogram.git"
```
Comment (sthagen) [inline package.json:18]: :+1:

**Review by sthagen — APPROVED** [review]
Thank you for executing the many ordering changes, introducing the new modules, and the update to a matching packaging git url.

**Comment by sthagen** [comment]
> Primarily tested the "Product Identification Helper" functionality. Could not identify any issues.
> 
> Could be helpful though for the end user to see examples of valid fields, especially for CPEs (instead of just the complex regex that needs to be satisfied). Examples would probably needed to be added to the CSAF JSON schema.

I was so confident of examples for CPEs inside the CSAF 2.0 prose / schema but ... we currently have not. I think in our discussions on the CPE regex we juggled with examples ...

I will consider adding such examples to the schema, thanks @pdamian

**Pull request merged**